app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/ProfileViewModel.kt

```diff
@@ -108,15 +111,18 @@ constructor(
    * @since 2.0
    * @last update 2.0
    */
-  fun retrieveTracks() {
-    viewModelScope.launch {
+  fun retrieveTracksFromSpotify(scope: CoroutineScope = CoroutineScope(Dispatchers.IO)) {
```

Comment:
I thought we decided to use the `viewModelScope` in the ViewModels, and only use `Dispatchers.IO` when necessary in the repository implementations. 
If we also want to inject scopes in the viewModel, I suggest doing so in the constructor, and setting the default value to `viewModelScope`

Reply:
I didn't do it but I can change it